scala/scala_2.13/src/main/scala/com/linecorp/armeria/scala/package.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria
+
+import com.google.common.util.concurrent.MoreExecutors
+import com.linecorp.armeria.common.annotation.UnstableApi
+
+import _root_.scala.concurrent.ExecutionContext
+
+/**
+ * Provides a collection of utilities for using Armeria in a project written in Scala.
+ * Read [[https://armeria.dev/docs/advanced-scala]] for more information.
+ */
+package object scala {
+
+  /**
+   * An `ExecutionContext` that runs the submitted `Runnable`s immediately on the thread that calls `execute()`
+   * and then yields back control to the caller after the `Runnable` finishes its execution.
+   *
+   * This is often useful when you need to avoid unnecessary context switches for the short-running tasks or
+   * asynchronous callbacks that are OK to run from an event loop thread.
+   *
+   * You must *not* use this `ExecutionContext` for a potentially long-running tasks.
+   */
+  @UnstableApi
+  val directExecutionContext: ExecutionContext = {
+    // Use ExecutionContext.parasitic which implements BatchingExecutor for Scala 2.13+.
+    val companionObjectClass =
+      Class.forName(s"${classOf[ExecutionContext].getName}$$", false, classOf[ExecutionContext].getClassLoader)
```

Comment:
What's the reason for using reflection? `ExecutionContext` already exists in 2.11 and will be used by any Future-based code (so there's no overhead for loading a new class that you're not going to use)

Reply:
Do you know how I can refer to the class of the `ExecutionContext` object in Scala? I just needed to retrieve `ExecutionContext$.parasitic` because it doesn't exist in Scala 2.12.

By the way, I actually was going to remove all this reflection code and just implement it without depending on `parasitic`, i.e. Let me remove this reflection anyway :laughing:

Reply:
`ExecutionContext.getClass` should be what you want

Reply:
No, wait, this *does* give you the same Class object as your `companionObjectClass` but both are wrong. There is no field or method called `parasitic` at all (in Scala 2.13.3). It is encoded directly as a static inner class. This is how you get the instance:

```
Class.forName(s"${classOf[ExecutionContext].getName}$$parasitic$$", false, classOf[ExecutionContext].getClassLoader)
  .getField("MODULE$").get(null)
  .asInstanceOf[ExecutionContext]
```

Reply:
JFYI, `ExecutionContext.parastic` could be accessed via `ExecutionContext.parastic$` 
That decompile to Java like:
```java
public interface ExecutionContext {
	public static class parasitic$ implements ExecutionContextExecutor, ... {
       ...
    }
}
```

Or use Scala reflection API? https://stackoverflow.com/a/63847599/1736581

Reply:
Thanks everyone for education.

Reply:
I replaced `directExecutionContext` with `ExecutionContexts.sameThread`. I also re-implemented it for simplicity.